src/map/utils/petutils.cpp

```diff
@@ -778,10 +779,18 @@ namespace petutils
             PPet->allegiance = PMaster->allegiance;
             PMaster->StatusEffectContainer->CopyConfrontationEffect(PPet);
 
-            PMaster->PPet = PPet;
             PPet->PMaster = PMaster;
 
-            PMaster->loc.zone->InsertPET(PPet);
+            if (PMaster->PInstance)
+            {
+                PPet->PInstance = PMaster->PInstance;
+                PMaster->PInstance->InsertPET(PPet);
```

Comment:
it looks like CZoneInstance already takes care of calling InsertPET on the right instance, so it should be possible to only set PPet->PInstance, and then do PMaster->loc.zone->InsertPET(PPet) for either way (eliminating the else, basically)

Reply:
ok I will test this in a bit

Reply:
yep, works